src/clj/puppetlabs/services/puppet_profiler/puppet_profiler_core.clj

```diff
@@ -72,6 +76,21 @@
      :mean mean
      :aggregate (* count mean)}))
 
+(defn puppetdb-metric
+  [[metric-name timer]]
+  (let [count (.getCount timer)
+        mean (metrics/mean-millis timer)
+        ;; Metric databases like Graphite and InfluxDB treat
+        ;; spaces and periods as delimiters.
+        ;;
+        ;; TODO: Should probably be abstracted into
+        ;; trapperkeeper-metrics as a general utility function.
+        munged-name (str/replace metric-name #"\.| " "_")]
```

Comment:
Catalog metrics are already outputting in this dot delineated way, right (ie `compiler.find_node`) (or are we munging those names in a different location that I'm missing)? Would this mean that similarly formatted metrics from the profiler will have different representations in the status endpoint? And if so, would it be better to have them all similarly formatted in the status endpoint (even if that means the PDB ones are less convenient) and shoot for adding a v2 that has a more convenient format for all the metrics?

Reply:
Yeah, but we're already doing transformation there. I.E `catalog.find_node` is shortened to `find_node` by the regex captures in `MetricsPuppetProfiler.java`. The additional work done here by munging the PuppetDB metrics is to reconcile the fact that the way we've instrumented the compiler Ruby code with`Puppet::Util::Profiler.profile` resulted in a very flat namespace:

  - `compiler.find_node`
  - `compiler.compile`
  - `compiler. static_compile`
  - etc.

But PuppetDB has ended up with a deeply nested namespace combined with odd spacing:

  - `puppetdb.facts.encode`
  - `puppetdb.command.submit.replace facts`
  - `puppetdb.catalog.munge`
  - `puppetdb.command.submit.replace catalog`

There's already a lot of drift between how things are named in the `/status` API and the `/metrics` API. The munging above makes the formatting of names in the `puppetdb-metrics` section of the `/status` API consistent with other sections like `catalog-metrics` which I think is more important than trying to keep it consistent with `/metrics`.